test/Microsoft.Azure.Devices.Edge.Test.Common/certs/CertificateAuthority.cs

```diff
@@ -2,48 +2,69 @@
 namespace Microsoft.Azure.Devices.Edge.Test.Common.Certs
 {
     using System;
+    using System.IO;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
     using RootCaKeys = System.ValueTuple<string, string, string>;
 
     public class CertificateAuthority
     {
         readonly Option<string> scriptPath;
 
-        public EdgeCertificates Certificates { get; }
+        // TODO: EdgeCertificate needs to be moved into certificate types.
+        public CaCertificates EdgeCertificate { get; set; }
 
         public static async Task<CertificateAuthority> CreateAsync(string deviceId, RootCaKeys rootCa, string scriptPath, CancellationToken token)
         {
-            (string rootCertificate, string rootPrivateKey, string rootPassword) = rootCa;
-            await OsPlatform.Current.InstallRootCertificateAsync(rootCertificate, rootPrivateKey, rootPassword, scriptPath, token);
-            EdgeCertificates certs = await OsPlatform.Current.GenerateEdgeCertificatesAsync(deviceId, scriptPath, token);
-            return new CertificateAuthority(certs, scriptPath);
+            if (!File.Exists(Path.Combine(scriptPath, FixedPaths.RootCaCert.Cert)))
+            {
+                // [9/26/2019] Setup all the environment for the certificates.
+                // More info: /iotedge/scripts/linux/runE2ETest.sh : run_test()
+                Log.Verbose("----------------------------------------");
+                Log.Verbose("Install Root Certificate");
+                (string rootCertificate, string rootPrivateKey, string rootPassword) = rootCa;
+                await OsPlatform.Current.InstallRootCertificateAsync(rootCertificate, rootPrivateKey, rootPassword, scriptPath, token);
+            }
+            else
+            {
+                Log.Verbose("----------------------------------------");
+                Log.Verbose("Skip Root Certificate: Root Certificate is installed");
+                Log.Verbose("----------------------------------------");
+            }
+
+            return new CertificateAuthority(scriptPath);
         }
 
         public static CertificateAuthority GetQuickstart()
         {
-            EdgeCertificates certs = OsPlatform.Current.GetEdgeQuickstartCertificates();
+            CaCertificates certs = OsPlatform.Current.GetEdgeQuickstartCertificates();
             return new CertificateAuthority(certs);
         }
 
-        CertificateAuthority(EdgeCertificates certs)
+        CertificateAuthority(string scriptPath)
         {
-            this.Certificates = certs;
-            this.scriptPath = Option.None<string>();
+            this.scriptPath = Option.Maybe(scriptPath);
         }
 
-        CertificateAuthority(EdgeCertificates certs, string scriptPath)
+        CertificateAuthority(CaCertificates certs)
```

Comment:
`this.scriptPath` needs to be set to `Option.None<string>()` explicitly in this constructor, otherwise methods `GenerateIdentityCertificatesAsync` and `GenerateCaCertificatesAsync` will throw a null reference exception.